### PR TITLE
Add pid to created and updated resource

### PR DESCRIPTION
**Pull request by fil512**
add RESOURCE_PID to resource UserData before STORAGE_PRECOMMIT_RESOURCE_CREATED and STORAGE_PRECOMMIT_RESOURCE_UPDATED hooks are called

### CODE REVIEW

**Comment by fil512** [comment]
@jamesagnew this is ready for review

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java [hunk 1]
```diff
@@ -1234,6 +1237,12 @@ public void beforeCommit(boolean readOnly) {
 		return savedEntity;
 	}
 
+	protected void addPidToResource(ResourceTable theEntity, IBaseResource theResource) {
+		if (theResource instanceof IAnyResource) {
+			IDao.RESOURCE_PID.put((IAnyResource) theResource, theEntity.getId());
+		}
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java]: This should also have an
```java
if (theResource instanceof IResource) {
   .....
}
```

..to cover the DSTU2 structures.

Comment (fil512) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java]: `UserData` isn't available on `IResource`.

Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java]:
```java
		IDao.RESOURCE_PID.put((IResource)resource, theEntity.getId());
```

Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java]: Sorry- you actually need to do both. Heh.

The DSTU2 structures only implement IResource, and the DSTU3+ structures only implement IAnyResource. This is legacy baggage.. :)

**Comment by fil512** [comment]
Ya I just noticed that when it stopped working :-)

**Pull request merged**